components/campaign/CampaignChaosBagView/index.tsx

```diff
@@ -0,0 +1,382 @@
+import React from 'react';
+import { keys, values } from 'lodash';
+import { BackHandler, Button, Platform, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { EventSubscription, Navigation } from 'react-native-navigation';
+
+import { t } from 'ttag';
+import { iconsMap } from '../../../app/NavIcons';
+import { NavigationProps } from '../../types';
+import { CHAOS_TOKENS, ChaosBag, ChaosTokenType } from '../../../constants';
+import { COLORS } from '../../../styles/colors';
+import { ChaosBagResults } from '../../../actions/types';
+import typography from '../../../styles/typography';
+import { EditChaosBagProps } from '../EditChaosBagDialog';
+import ChaosToken from '../ChaosToken';
+
+export interface CampaignChaosBagProps {
+    chaosBag: ChaosBag;
+    chaosBagResults: ChaosBagResults;
+    updateChaosBag: (chaosBag: ChaosBag) => void;
+    trackDeltas?: boolean;
+}
+
+interface State {
+    chaosBagResults: ChaosBagResults;
+    visible: boolean;
+    iconKey?: ChaosTokenType;
+}
+
+type Props = CampaignChaosBagProps & NavigationProps;
+
+// TODO: need to change it to pull chaosBag from redux state (pass the campaignId, and fetch campaign + get it from redux)
+export default class CampaignChaosBagView extends React.Component<Props, State> {
+  static get options() {
+    return {
+      topBar: {
+        leftButtons: [
+          Platform.OS === 'ios' ? {
+            systemItem: 'cancel',
+            text: t`Cancel`,
+            id: 'back',
+            color: COLORS.navButton,
+            testID: t`Cancel`,
+          } : {
+            icon: iconsMap['arrow-left'],
+            id: 'androidBack',
+            color: COLORS.navButton,
+            testID: t`Back`,
+          },
+        ],
+        rightButtons: [{
+          systemItem: 'save',
+          text: t`Edit`,
+          id: 'edit',
+          showAsAction: 'ifRoom',
+          color: COLORS.navButton,
+          testID: t`Edit`,
+        }],
+      },
+    };
+  }
+
+  _navEventListener?: EventSubscription;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      chaosBagResults: {
+        currentToken: undefined,
+        drawnTokens: [],
+        total: 0,
+      },
+      visible: true,
+    };
+
+    this._navEventListener = Navigation.events().bindComponent(this);
+  }
+
+  componentDidMount() {
```

Comment:
So the existing edit dialog did all this 'back button' override so there would be a confirmation when you press back.  I THINK this screen might be able to get away with the standard back functionality, since theoretically the set of drawn tokens + sealed tokens is 'non-destructive'?

What do you think?   If so, I think you can just remove these back button event listeners (as well as the handling of 'back' and 'androidBack') and just update the redux store on every draw/seal/clear?

Reply:
Yeah it it's non-destructive so that should be fine as long as the state of the drawn and sealed tokens persist.